pilot/cmd/pilot-agent/main.go

```diff
@@ -305,6 +295,39 @@ var (
 	}
 )
 
+func getPilotSAN(domain string, ns string) []string {
+	var pilotSAN []string
+	if controlPlaneAuthPolicy == meshconfig.AuthenticationPolicy_MUTUAL_TLS.String() {
+		pilotTrustDomain := role.TrustDomain
```

Comment:
I'm getting blind - where is 'role' defined and how is it set up ?

Reply:
Role is model.Proxy in main.go to store the config for Envoy, and here's the setup of role.TrustDomain:
https://github.com/istio/istio/pull/9059/files/61418c70f6426b1420d3c70e967ee56af6becbe5#diff-4f4201d9726cc132ac249be3818eb026R366

Reply:
Role is model.Proxy in main.go to store the config for Envoy, and here's the setup of role.TrustDomain:
https://github.com/istio/istio/pull/9059/files/61418c70f6426b1420d3c70e967ee56af6becbe5#diff-4f4201d9726cc132ac249be3818eb026R366